bionic/task_state.py

```diff
@@ -22,30 +26,23 @@ def __init__(self, task, dep_states, case_key, provider, entity_defs_by_dnode):
         self.provider = provider
         self.entity_defs_by_dnode = entity_defs_by_dnode
 
+        # Cached values.
+        self.task_keys = task.keys
         # In theory different entities for a single task could have different cache
         # settings, but I'm not sure it can happen in practice (given the way
         # grouped entities are created). At any rate, once we have tuple
         # descriptors, each task state will only be responsible for a single entity
         # and this won't be an issue.
-        can_persist, can_memoize = single_unique_element(
-            (entity_def.can_persist, entity_def.can_memoize)
-            for entity_def in entity_defs_by_dnode.values()
-        )
-
-        # Cached values.
-        self.task_keys = task.keys
-        self.should_memoize = can_memoize
-        self.should_persist = (
-            can_persist
-            and not self.task.is_simple_lookup
-            and not self.output_would_be_missing()
+        self.should_memoize = single_unique_element(
```

Comment:
Since this is now set at a different point from `should_persist`, I wonder if it should have a different name? It has basically the same semantics as `EntityDefinition.can_persist`, so maybe it could just be called `can_persist`?

Reply:
do you mean `can_memoize`?

Reply:
Assuming that's what you meant, I'm fine either way. I like that `should_memoize` is similar to `should_persist` and we can later change the logic for it without worrying about naming.

But I'm fine keeping it `can_memoize` for now and worry about that later. Basically following our philosophy of **err on the side of adopting feedback** (from go/trustme).

Reply:
Sorry, yes, I did mean `can_memoize`, not `can_persist`!

I also like that we have a separate layer for both `memoize` and `persist`, but since this PR makes them less analogous, it seems less attractive for them to have analogous names. (Unless we want to move `should_memoize` to be set during `initialize()` as well?)